compiler-core/src/ast.rs

```diff
@@ -302,6 +320,15 @@ pub enum Statement<T, Expr, ConstantRecordTag, PackageName> {
         doc: Option<String>,
     },
 
+    /// A named collections of keys and values
```

Comment:
```suggestion
    /// A newly defined type with one or more constructors.
    /// Each variant of the custom type can contain different types, so the type is
    /// the product of the types contained by each variant.
    ///
    /// This might be called an algebraic data type (ADT) or tagged union in other
    /// languages and type systems.
    ///
```